docs/advanced-guide/injecting-databases-drivers/page.md

```diff
@@ -712,4 +712,185 @@ func queryDataPoints(c *gofr.Context) (any, error) {
+	Select(ctx context.Context, table string) ([]map[string]interface{}, error)
+}
+
+```
+Import the gofr's external driver for SurrealDB:
+```go
+go get gofr.dev/pkg/gofr/datasource/surrealdb
+```
+The following example demonstrates injecting an SurrealDB instance into a GoFr application.
+
+```go
+package main
+
+import (
+	"gofr.dev/pkg/gofr"
+	"gofr.dev/pkg/gofr/datasource/surrealdb"
+)
+
+type Person struct {
+	ID    string `json:"id,omitempty"`
+	Name  string `json:"name"`
+	Age   int    `json:"age"`
+	Email string `json:"email,omitempty"`
+}
+
+func main() {
+	app := gofr.New()
+
+	client := surrealdb.New(surrealdb.Config{
+		Host:       "localhost",
+		Port:       8000,
+		Username:   "root",
+		Password:   "root",
+		Namespace:  "test_namespace",
+		Database:   "test_database",
+		TLSEnabled: false,
+	})
+
+	app.AddSurrealDB(client)
+
+	//GET
+	app.GET("/person", func(ctx *gofr.Context) (interface{}, error) {
+		persons, err := ctx.SurrealDB.Select(ctx, "person")
+		if err != nil {
+			ctx.Logger.Error("Select error: ", err)
+			return nil, err
+		}
+
+		if len(persons) == 0 {
+			return []interface{}{}, nil
+		}
+
+		return persons, nil
+	})
+
+	// POST
+	app.POST("/person", func(ctx *gofr.Context) (interface{}, error) {
+		var person Person
+		if err := ctx.Bind(&person); err != nil {
+			ctx.Logger.Error("Binding error: ", err)
+			return nil, err
+		}
+
+		query := "CREATE person SET name = $name, age = $age, email = $email;"
+		vars := map[string]interface{}{
+			"name":  person.Name,
+			"age":   person.Age,
+			"email": person.Email,
+		}
+
+		ctx.Logger.Debugf("Executing query: %s with vars: %+v", query, vars)
+
+		result, err := ctx.SurrealDB.Query(ctx, query, vars)
+		if err != nil {
+			ctx.Logger.Errorf("Query error: %v", err)
+			return nil, err
+		}
+
+		ctx.Logger.Debugf("Raw result: %+v", result)
+
+		verifyResult, err := ctx.SurrealDB.Query(ctx, "SELECT * FROM person ORDER BY id DESC LIMIT 1;", nil)
+		if err != nil {
+			ctx.Logger.Errorf("Verify error: %v", err)
+		} else {
+			ctx.Logger.Debugf("Verify result: %+v", verifyResult)
+		}
+
+		return map[string]interface{}{
+			"message": "Creation attempted",
+			"result":  result,
+			"verify":  verifyResult,
+		}, nil
+	})
+
+	// PUT
+	app.PUT("/person/{id}", func(ctx *gofr.Context) (interface{}, error) {
+		id := ctx.PathParam("id")
+		var person Person
+		if err := ctx.Bind(&person); err != nil {
+			ctx.Logger.Error("Binding error: ", err)
+			return nil, err
+		}
+
+		query := "UPDATE person:" + id + " SET name = $name, age = $age, email = $email RETURN *"
+		vars := map[string]interface{}{
+			"name":  person.Name,
+			"age":   person.Age,
+			"email": person.Email,
+		}
+
+		result, err := ctx.SurrealDB.Query(ctx, query, vars)
+		if err != nil {
+			ctx.Logger.Error("Update error: ", err)
+			return nil, err
+		}
+
+		if len(result) > 0 && len(result[0].(map[interface{}]interface{})) > 0 {
+			return result[0].(map[interface{}]interface{}), nil
+		}
```

Comment:
I would replace the `interface{}` notation with `any` it would be more readable especially here